src/sentry/utils/email.py

```diff
@@ -109,7 +114,15 @@ def make_msgid(domain):
     return msgid
 
 
-FROM_EMAIL_DOMAIN = domain_from_email(settings.DEFAULT_FROM_EMAIL)
+# cache the domain_from_email calculation
+_from_email_domain_cache = [None, None]
+
+
+def get_from_email_domain():
+    from_ = options.get('mail.from')
+    if not _from_email_domain_cache[0] == from_:
+        _from_email_domain_cache[0] = domain_from_email(from_)
+    return _from_email_domain_cache[1]
```

Comment:
Going to redo this because I straight up don't remember, and this is obviously wrong. Will add tests too.